bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/form/FormPostServiceImpl.java

```diff
@@ -0,0 +1,191 @@
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.osgi.services.HttpClientBuilderFactory;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.servlets.post.JSONResponse;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.wcm.core.components.services.form.FormPostService;
+import com.day.cq.wcm.foundation.forms.FormsHelper;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+@Component(
+        service = {FormPostService.class}
+)
+@Designate(
+        ocd = FormPostServiceImpl.Config.class
+)
+public class FormPostServiceImpl implements FormPostService {
+
+    private static final int DEFAULT_CONNECTION_TIMEOUT = 6000;
+    private static final int DEFAULT_SOCKET_TIMEOUT = 6000;
+    private static final Logger LOG = LoggerFactory.getLogger(FormPostServiceImpl.class);
+    private static final Set<String> INTERNAL_PARAMETER = ImmutableSet.of(
+            ":formstart",
+            "_charset_",
+            ":redirect",
+            ":cq_csrf_token"
+    );
+
+    private static final String PN_FORM_ENDPOINT_URL = "formEndPointUrl";
+    private static final String CHARSET = "UTF-8";
+
+    private int connectionTimeout;
+    private int socketTimeout;
+
+    @Reference
+    private HttpClientBuilderFactory clientBuilderFactory;
+
+    @Override
+    public boolean sendFormData(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+        try {
+            JSONObject jsonRequest = requestParamsToJSON(request);
+            return processFormApi(request, response, jsonRequest);
+        } catch (JSONException e) {
+            LOG.error(e.getMessage());
+        }
+        return false;
+    }
+
+    private boolean processFormApi(SlingHttpServletRequest request, SlingHttpServletResponse response, JSONObject jsonRequest) {
```

Comment:
I don't like to have `api` in the method name, I'd call it `processFormData` instead.